toolchain/check/testdata/impl/lookup/min_prelude/find_in_final.carbon

```diff
@@ -20,16 +20,31 @@ interface I {
 final impl forall [U:! type] U as I where .T = () {}
 
 fn F(V:! I) -> V.T {
-  // TODO: Even though we have a witness that `V impls I` from the constraint
-  // on `I`, we should do an impl lookup to see if any effectively final impl
-  // applies when we find an unknown value in that witness. In this case, that
-  // lookup would find an impl with more specific values for associated
-  // constants that we should merge.
+  // Even though we have a witness that `V impls I` from the constraint on `I`,
+  // we should do an impl lookup to see if any effectively final impl applies
+  // when we find an unknown value in that witness. In this case, that lookup
+  // would find an impl with more specific values for associated constants that
+  // we should merge.
+  return ();
+}
+
+// --- fail_todo_use_where_in_facet_value.carbon
+library "[[@TEST_NAME]]";
+
+interface I {
+  let T:! type;
+}
+
+final impl forall [U:! type] U as I where .T = {.r: ()} {}
 
-  // CHECK:STDERR: fail_todo_keep_looking_for_final_impl.carbon:[[@LINE+7]]:3: error: cannot implicitly convert expression of type `()` to `V.(I.T)` [ConversionFailure]
+fn F(V:! I where .T = ()) -> V.T {
```

Comment:
What an interesting testcase! I think I'm inclined to say that this `where` expression should be invalid, perhaps because the left-hand side of the `=` constraint is not symbolic (`.T` is `.Self.T` which is `{.r: ()}`). Not sure exactly what rule to use, but this can locally be determined to be an impossible constraint, so I think we ought to reject for that reason at least.

Reply:
Looks like I had two tests doing roughly the same thing, combined them and made it a `todo_fail_`.